serializations.md

```diff
@@ -0,0 +1,321 @@
+# Serialization formats
+
+The serialization formats in this document describes the binary format
+used to encode the different objects in the binary format that is used
+for hashing. Other formats may be used for communication between
+nodes.
+
+## Static size object serialization
+
+#### Block header
+
+For block headers all fields sizes are statically known and can be
+constructed directly as a byte array.
+
+| Fieldname | Size (bytes) |
+| --- | --- |
+| height       | 8    |
+| prev_hash    | 32   |
+| txs_hash     | 32   |
+| root_hash    | 32   |
+| target       | 8    |
+| nonce        | 8    |
+| time         | 8    |
+| version      | 8    |
```

Comment:
Wouldn't it make sense to have version as the first field to make it possible to change the field after version in future versions?

Reply:
That is if we want to be able to reuse the hashing serialization for sending header and blocks to other nodes, which I think would be a good idea.

Reply:
That would make sense. This part is only copied from the current documentation, but we could change this as well.

Reply:
Version is already the first in the code https://github.com/aeternity/epoch/blob/v0.9.0/apps/aecore/src/aec_headers.erl#L109